backend/algorithms/objects/conditions.py

```diff
@@ -109,20 +111,29 @@ def set_logic(self, logic: Logic):
 
     def validate(self, user: User) -> tuple[bool, list[str]]:
         """ Returns True if the user is taking these courses in the same term """
+        
         match self.logic:
             case Logic.AND:
-                return all(
+                met = all(
                     user.has_taken_course(course)
                     or user.is_taking_course(course)
                     for course in self.courses
-                ), []
+                )
+                warning = []
+                for course in self.courses:
+                    if not (user.has_taken_course(course)
+                    or user.is_taking_course(course)) :
+                        warning.append(course)
+                return met, ([] if met else ['(Corequisites: ' + ' AND '.join(warning) + ')'])
             case Logic.OR:
-                return any(
+                met = any(
```

Comment:
really minor nitpick but this name could perhaps be a bit more descriptive;
Currently, not sure if it's conditions met or courses w/ conditions met